### PR TITLE
Integrate new PT routing algorithm

### PR DESCRIPTION
**Pull request by mark-idleman**
Encompasses all changes involved in integrating the new transit routing algorithm Michael has been working on. High-level breakdown of changes:
- Point our GH dependency to the latest commit in the [branch](https://github.com/graphhopper/graphhopper/commits/patterns) Michael's created for developing the new algo
- Updates test configs to include new necessary fields for "turning on" the new algo, and updates unit tests accordingly, as the new algo results in slightly different transit route responses
- Updates several spots in custom Replica-specific code to use the new `PtRouterTripBasedImpl`, necessary for the new algo to actually get used in route responses
- Many small changes (especially in `CustomOsmReader`) simply to catch up code we've copy/pasted from GH master with the newer GH master version Michael's built his branch off of

This is the main branch we've been testing off of, so the conclusions about performance/quality we've laid out in the #prj-new-transit-algorithm slack channel cover the basics of how this new algo performs in production settings.

### CODE REVIEW

**Review by bryanwilliams1025 — APPROVED** [review]
Exiting! Looks good, just a couple general thoughts:

- Any update on getting @michaz 's new PT algo branch merged into core Graphhopper? I see the unit tests on the branch are currently failing. Agree we don't need to block on it, but it would make me nervous to remain on that branch indefinitely. If we let it sit, getting back to core GH master eventually could be painful. Just want to make sure this piece doesn't get lost. cc @kaelgreco 
- Are we ready to merge the model-side PR too? Happy to review that PR even before this one merges. Ideally we'd merge the two PRs in quick succession; otherwise, any Graphhopper changes we need in model would be blocked on merging the new algo. Realize we don't have any major router work planned besides getting the new algo merged, but I could see minor stuff popping up (HNTB tweaks including GUI improvements, FT fixes, etc.), so just wanted to flag.

**Comment by mark-idleman** [comment]
Thanks for the quick review @bryanwilliams1025 , and great questions!

- No changes yet on the status of Michael's branch, but it's not something we're going to wait long on. I just pushed the issue [here](https://replicahq.slack.com/archives/CK358RL0Z/p1699561825118729) so hopefully we get clarity soon; a fast-follow PR will be put up right away once Michael's main branch has been merged
- Yes, the model-side PR is almost ready, I'm just finishing cleanup there. Good point that staggering the merges too much could lead to trouble; I'm happy to get them both to ✅ and then merging in fast succession to minimize issues! I should have the other PR ready for review ~EOD today

One other side note - I had a realization that the new GTFS validator image I created doesn't actually use the new PT algo, because we hadn't set the proper flag + set up a mechanism for exact GTFS schedule days to be passed into the GH config used for validation. I just pushed the fix with db65db1, and will build a new validator image + add/test the matching model repo-side logic before I merge this PR

**Comment by mark-idleman** [comment]
Phew, at long last, I figured out the weird java update/guice/etc issue! At the end of the day, all that was needed was:
- bumping `guice` to the latest version, `7.0.0`
- adding a new dep for `jakarta.inject-api` (the error I had after bumping guice was related to jakarta classes not being found)
- bumping `guava` to the latest version, so it didn't clash with the bumped versions of ^

I built a MNC router via arti using the image resulting from the latest commit on this branch, and spun it up for testing - the GUI worked as-expected and there were no errors in the logs 🎉

**Pull request merged**
